packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -91,6 +91,20 @@ export interface EcsRunTaskProps extends sfn.TaskStateBaseProps {
    * @default false
    */
   readonly enableExecuteCommand?: boolean;
+
+  /**
+   * Cpu setting override
+   *
+   * @default - No override
+   */
+  readonly cpu?: string;
```

Comment:
Why not use number type to be consistent with [other similar props](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_stepfunctions_tasks.ContainerOverride.html#cpu)?

Reply:
@tmokmss 
In the AWS API and task definitions, cpu is string type, and changing it to number type may cause unexpected behavior.

https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts#L157

Reply:
> may cause unexpected behavior.
Not really, we can just stringfy it after receiving the value. All the existing code will remain unchanged.

Then, the problem is whether we should use string or number. I'm not sure why the current code is inconsistent (e.g. TaskDefinition uses string, while FargateTaskDefinition uses number). Maybe looking through git history will help to find why. If there is no particular reason, I'd prefer number type because the existing ContainerOverride interface also uses number, which is closer to this interface than TaskDefinition construct.

Reply:
All right. I'll fix ix.

Reply:
@tmokmss 
I tried to fix it, but simply changing to number type failed the integration test. 
```
Failed resources:
aws-sfn-tasks-ecs-fargate-run-task | 5:48:49 PM | UPDATE_FAILED        | AWS::StepFunctions::StateMachine      | StateMachine (StateMachine2E01A3A5) Resource handler returned message: "Invalid State Machine Definition: 'SCHEMA_VALIDATION_FAILED: The value for the field 'Cpu' must be a STRING at /States/FargateTask/Parameters' (Service: AWSStepFunctions; Status Code: 400; Error Code: InvalidDefinition; Request ID: ed6dadc7-f05a-474e-abfa-bc5291934ab8; Proxy: null)" (RequestToken: e1181314-190c-640a-43ca-ec6ccd123530, HandlerErrorCode: InvalidRequest)
```

Where in the code do you think we should stringfy this value?

Reply:
done 7061480

Reply:
renderOverrides is where CFn template is actually rendered, so you can stringify the values there. To stringify the value, I'd recommend to look at the existing code such as FargateContainerDefinition. You have to take care of CDK tokens.

Reply:
Sorry, I now find that they are already inconsistent at CFn level... [ContainerOverride](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ContainerOverride.html) uses integer as cpu while [TaskOverride](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_TaskOverride.html) uses string as cpu. Given that, I think it's better to just follow CFn type definitions, because in the future CFn may start to use an actual string that cannot be parsed to a number. What do you think? The below API is so weird but we won't override two cpu props at once anyway...

```ts
const runTask = new tasks.EcsRunTask(this, 'Run', {
  // ...
  cpu: '1024',
  containerOverrides: [
    cpu: 1024,
  ]
});
```

Reply:
We cannot change the existing type definition, so we have to choose which one is better:

```ts
// option 1: weird but maybe safer
const runTask = new tasks.EcsRunTask(this, 'Run', {
  // ...
  cpu: '1024',
  containerOverrides: [
    cpu: 1024,
  ]
});

// option 2: more natural, but prone to future breaking changes
const runTask = new tasks.EcsRunTask(this, 'Run', {
  // ...
  cpu: 1024,
  containerOverrides: [
    cpu: 1024,
  ]
});
```

I now prefer option 1, but anyway I'll approve the review once the all the PR check passes, and let the core team decide it. Any thoughts?

Reply:
Done (option 1) [8c79da9](https://github.com/aws/aws-cdk/pull/30140/commits/8c79da968b328cf4429668afccfd5ed416edbb36)